impl/src/main/java/com/okta/sdk/impl/resource/DefaultApplicationBuilder.java

```diff
@@ -65,6 +65,10 @@ public T setSelfService(Boolean selfService) {
 
     @Override
     public T setSignOnMode(ApplicationSignOnMode signOnMode) {
+        if(signOnMode == ApplicationSignOnMode.SDK_UNKNOWN) {
+            throw new IllegalArgumentException(
+                "The " + signOnMode.getClass().getName() + ".SDK_UNKNOWN can not be used in setter");
+        }
```

Comment:
If this logic wasn't here would the above `setProperty` logic get triggered?

Reply:
No. The `DefaultApplicationBuilder` class doesn't extend the abstract `AbstractPropertyRetriever` class. That is why we need this logic here.